src/mame/drivers/jpmsru.cpp

```diff
@@ -926,6 +1031,55 @@ static INPUT_PORTS_START( j_la )
 	PORT_BIT( 0x80, IP_ACTIVE_HIGH, IPT_COIN4 ) PORT_NAME("1p")
 INPUT_PORTS_END
 
+static INPUT_PORTS_START( j_lc )
+	PORT_INCLUDE( jpmsru_inputs )
+
+	PORT_MODIFY("IN0")
+	PORT_CONFNAME( 0x80, 0x80, "Game type" )
+	PORT_CONFSETTING(    0x00, "Club Casino (£50, 5p stake)" )
+	PORT_CONFSETTING(    0x80, "Lucky Casino (£100, 10p stake)" )
+
+	PORT_MODIFY("IN1")
+	PORT_BIT( 0x01, IP_ACTIVE_HIGH, IPT_POKER_HOLD1 ) PORT_NAME("Hold 1")
+	PORT_BIT( 0x02, IP_ACTIVE_HIGH, IPT_POKER_HOLD2 ) PORT_NAME("Hold 2")
+	PORT_BIT( 0x04, IP_ACTIVE_HIGH, IPT_POKER_HOLD3 ) PORT_NAME("Hold 3")
+	PORT_BIT( 0x08, IP_ACTIVE_HIGH, IPT_POKER_HOLD4 ) PORT_NAME("Hold 4")
+	PORT_BIT( 0x10, IP_ACTIVE_HIGH, IPT_SERVICE ) PORT_NAME("Self Test")
+	PORT_BIT( 0x20, IP_ACTIVE_HIGH, IPT_POKER_CANCEL ) PORT_NAME("Cancel")
+	PORT_BIT( 0x40, IP_ACTIVE_HIGH, IPT_START1 ) PORT_NAME("Start")
+	PORT_DIPNAME( 0x80, 0x00, "Win Freeze" )
+	PORT_DIPSETTING(    0x00, DEF_STR(Off) )
+	PORT_DIPSETTING(    0x80, DEF_STR(On) )
+
+	PORT_MODIFY("IN2")
+	PORT_DIPNAME( 0x01, 0x00, "Test Hold" ) // Always gives holds if door is open
+	PORT_DIPSETTING(    0x00, DEF_STR(Off) )
+	PORT_DIPSETTING(    0x01, DEF_STR(On) )
+	PORT_BIT( 0x02, IP_ACTIVE_HIGH, IPT_BUTTON1 ) PORT_NAME("Take Win")
+	PORT_BIT( 0x04, IP_ACTIVE_HIGH, IPT_BUTTON2 ) PORT_NAME("Take Half")
```

Comment:
There are `IPT_GAMBLE_TAKE` and `IPT_GAMBLE_HALF` types for these.

Reply:
Fixed.